core/server/proxy/src/main/java/alluxio/proxy/s3/ListBucketResult.java

```diff
@@ -0,0 +1,404 @@
+import java.util.List;
+
+/**
+ * Get bucket result defined in http://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html
+ * It will be encoded into an XML string to be returned as a response for the REST call.
+ *
+ * TODO(chaomin): consider add more required fields in S3 BucketGet API.
+ */
+@JacksonXmlRootElement(localName = "ListBucketResult")
+@JsonPropertyOrder({ "Name", "Prefix", "ContinuationToken", "NextContinuationToken",
+    "KeyCount", "MaxKeys", "IsTruncated", "Contents" })
+public class ListBucketResult {
+
+  /* Name of the bucket. */
+  private String mName = "";
+  /* Keys that begin with the indicated prefix. */
+  private String mPrefix = null;
+  /* ContinuationToken is included in the response if it was sent with the request. */
+  private String mContinuationToken = null;
+  /**
+   * Returns the number of keys included in the response. The value is always less than or equal
+   * to the mMaxKeys value.
+   */
+  private int mKeyCount = 0;
+  /* The maximum number of keys returned in the response body. */
+  private int mMaxKeys = S3Constants.S3_DEFAULT_MAX_KEYS;
+  /**
+   * Specifies whether or not all of the results were returned. If the number of
+   * results exceeds that specified by MaxKeys, all of the results might not be returned.
+   */
+  private boolean mIsTruncated = false;
+  /**
+   * If only partial results are returned (i.e. mIsTruncated = true), this value is set as the
+   * next continuation token. The continuation token is the full Alluxio path of the object.
+   */
+  private String mNextContinuationToken = null;
+  /* A list of objects with metadata. */
+  private List<Content> mContents = new ArrayList<>();
+
+  /**
+   * Creates an {@link ListBucketResult}.
+   */
+  public ListBucketResult() {}
+
+  /**
+   * Creates an {@link ListBucketResult}.
+   *
+   * @param bucketName the bucket name
+   * @param objectsList a list of {@link URIStatus}, representing the objects
+   * @param options the list bucket options
+   */
+  public ListBucketResult(
+      String bucketName, List<URIStatus> objectsList, ListBucketOptions options) {
+    mName = bucketName;
+    mPrefix = options.getPrefix();
+    mKeyCount = 0;
+    if (options.getMaxKeys() != null) {
+      mMaxKeys = Integer.parseInt(options.getMaxKeys());
+    }
+    mContents = new ArrayList<>();
+    mContinuationToken = options.getContinuationToken();
+
+    Collections.sort(objectsList, new URIStatusComparator());
+
+    int startIndex = 0;
+    if (options.getContinuationToken() != null) {
+      URIStatus tokenStatus = new URIStatus(new FileInfo().setPath(
+          mName + AlluxioURI.SEPARATOR + mContinuationToken));
+      startIndex = Collections.binarySearch(objectsList, tokenStatus, new URIStatusComparator());
+      if (startIndex < 0) {
+        // If continuation token does not exist in the object list, find the first element which is
+        // greater than the token.
+        startIndex = (-1) * startIndex - 1;
```

Comment:
nit: do we need parentheses around -1?

Reply:
I slightly prefer parentheses `(-1)` in case we add more math around it :)